protocol/protocol.proto

```diff
@@ -144,6 +144,9 @@ message MiniblockHeader {
     // pointer to block with previous snapshot
     int64 prev_snapshot_miniblock_num = 7;
 
+    // hash of the snapshot.
+    bytes snapshot_hash = 8;
+
```

Comment:
can we make this optional? I realize a 0 byte snapshot is essentially nil, but optional makes the frontend code much more clear what to expect.